dashboard/src/app/[tenantId]/components/CopyButton.tsx

```diff
@@ -0,0 +1,29 @@
+'use client'
+import { FC, useState } from 'react'
+import { Button } from '@/components/ui/button'
+import { Check, Copy } from 'lucide-react'
+import { cn } from '@/components/utils'
+
+interface CopyButtonProps {
+  value: string
+  className?: string
+}
+
+export const CopyButton: FC<CopyButtonProps> = ({ value, className }) => {
+  const [copied, setCopied] = useState(false)
+
+  return (
+    <Button
+      variant="ghost"
+      size="icon"
+      onClick={() => {
+        navigator.clipboard.writeText(value)
+        setCopied(true)
+        setTimeout(() => setCopied(false), 1000)
```

Comment:
Minor issue with this is that the state of the timeout isn't saved. This is not a big deal but it will technically cause UI issues if the user try's to copy it more than once in 1 second.

I'm sure there is a hook that better handles setTimeout

Reply:
technically perhaps, but honestly i think it's fine:

https://github.com/user-attachments/assets/c0104dc1-c191-4a23-a21d-46cdd869e69c